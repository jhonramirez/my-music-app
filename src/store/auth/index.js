export * from "./authSlice";
export * from "./playlistSlice";