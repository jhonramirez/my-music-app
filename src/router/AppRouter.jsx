import { Routes, Route } from 'react-router-dom';
import { LoginPage } from '../auth';
import { MusicRoutes } from '../music';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

export const AppRouter = () => {
    return (
        <>
            <Routes>
                <Route path="login" element={
                    <PublicRoute>
                        <LoginPage />
                    </PublicRoute>} />
                <Route path="/*" element={
                    <PrivateRoute>
                        <MusicRoutes />
                    </PrivateRoute>} />
            </Routes>
        </>
    )
}
