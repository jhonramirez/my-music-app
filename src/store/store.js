import { configureStore } from "@reduxjs/toolkit";
import { authSlice, playlistSlice } from "./auth";

export const store = configureStore({
  reducer: {
    auth: authSlice.reducer,
    playlist: playlistSlice.reducer,
  },
});
