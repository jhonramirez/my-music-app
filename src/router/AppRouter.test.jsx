import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { authSlice, playlistSlice } from "../store";
import { AppRouter } from "./AppRouter";
import { MemoryRouter } from "react-router-dom";

jest.mock("../services/createDB");

const store = configureStore({
  reducer: {
    auth: authSlice.reducer,
    playlist: playlistSlice.reducer,
  },
});

describe("Pruebas en <MusicRoutes/>", () => {
  const { container } = render(
    <Provider store={store}>
      <MemoryRouter>
        <AppRouter />
      </MemoryRouter>
    </Provider>
  );

  test("Debe hacer match con el snapshot", () => {
    expect(container).toMatchSnapshot();
  });
});
