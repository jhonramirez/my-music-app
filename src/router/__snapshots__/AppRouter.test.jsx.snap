// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MusicRoutes/> Debe hacer match con el snapshot 1`] = `
<div>
  <section>
    <article
      class="loginPage"
    >
      <img
        alt="My music App"
        class="loginPage__img"
        src="Mymusicapp.png"
      />
      <h1>
        My music App
      </h1>
      <hr />
      <button
        class="btn btn__black"
        data-testid="button"
      >
        Login
      </button>
    </article>
  </section>
</div>
`;
